src/primer_filter.cpp

```diff
@@ -17,100 +17,94 @@ PrimerFinder::~PrimerFinder() {
     // nothing to do
 }
 
-const vector<PrimerPair>& PrimerFinder::get_primer_pairs() const {
-    return primer_pairs;
+const vector<PrimerPair>& PrimerFinder::get_primer_pairs_of_chrom(const string chrom_name) const {
+    return chroms.at(chrom_name);
 }
 
-const vector<PrimerPair>& PrimerFinder::get_selected_primer_pairs() const {
-    return selected_primer_pairs;
-}
-
-
 // Make a new pair of primers with given attributes. Primers are processed and 
 // added to primer_pairs and selected_primer_pairs.
-void PrimerFinder::add_primer_pair(const size_t& left_primer_starting_node_id,
-        const size_t& left_primer_offset, const size_t& left_primer_length,
-        const size_t& right_primer_starting_node_id,
-        const size_t& right_primer_offset, const size_t& right_primer_length) {
+void PrimerFinder::add_primer_pair(const string& path_name,
+    const size_t& left_primer_starting_node_id, const size_t& left_primer_offset,
+    const size_t& left_primer_length, const size_t& right_primer_starting_node_id,
+    const size_t& right_primer_offset, const size_t& right_primer_length) {
     
-    primer_pairs.emplace_back();
-    PrimerPair& primer_pair = primer_pairs.back();
+    chroms.at(path_name).emplace_back();
+    PrimerPair& primer_pair = chroms.at(path_name).back();
     primer_pair.right_primer.left = false;
 
-    make_primer(primer_pair.left_primer, left_primer_starting_node_id, left_primer_offset, left_primer_length, true);
-    make_primer(primer_pair.right_primer, right_primer_starting_node_id, right_primer_offset, right_primer_length, false);
+    make_primer(primer_pair.left_primer, path_name, left_primer_starting_node_id, left_primer_offset, left_primer_length, true);
+    make_primer(primer_pair.right_primer, path_name, right_primer_starting_node_id, right_primer_offset, right_primer_length, false);
     primer_pair.linear_product_size = primer_pair.right_primer.position - primer_pair.left_primer.position + primer_pair.right_primer.length;
     update_min_max_product_size(primer_pair);
-    if (no_variation(primer_pair)) {
-        primer_pair.no_variation = true;
-        selected_primer_pairs.push_back(primer_pairs.back());
-    }
-
+    update_variation(primer_pair, path_name);
 }
 
 void PrimerFinder::load_primers(const string& path_to_primers) {
```

Comment:
Not a huge deal, but a more flexible pattern is to take an `istream&` as the input to the function rather than a filepath. That way the calling code can make its own `ifstream` by opening the file or alternatively substitute in `cin` or a `stringstream`.